package/libpiksi/libpiksi/src/settings.c

```diff
@@ -1265,6 +227,8 @@ void settings_destroy(settings_ctx_t **ctx)
     return;
   }
 
+  setreg_destroy(&(*ctx)->setreg);
```

Comment:
I'm not sure I follow there relation of `destroy` and `settings_destroy`, but it would be best if there was a single clearly defined way of destroying a `settings_ctx_t` safely.

Reply:
This was based on the legacy code. Streamlined it a bit.